specifications/Dutch Auction.md

```diff
@@ -0,0 +1,172 @@
+The price curve should take into account the need for new validator in both the starting price and the rate of change.
+
+#### Context
+AliceNet validator nodes communicate with each other in a secure way through the use of a distributed key.
+The operation through which this distributed key is generated is called ETHDKG (Ethereum Distribute Key Generation) ceremony.
+
+When a new validator is admitted to join AliceNet, two ETHDKG ceremonies must be executed:
+* One to destroy the current distributed key (all current validators)
+* One to generate a new distributed key (all current validators plus the new one)
+
+Since this operation involves interacting with multiple Ethereum Smart Contract functions it is a costly operation estimated in approx 2.5M gas units for each validator.
+To encourage current validators incur in this expense whenever a new validator wants to join the network, an amount to be paid by the new validator will be required.
+This amount will be determined by an auction among the postulants, the one that wins pays the price and is allowed to join the network, the minimum price for this auction must cover at least the expenses of all current validators.
+
+Regular auction systems need to maintain the auction open for a fixed period of time, adding complexity and creating sometimes unnecessary waits. There is a simpler and faster auction system called Dutch Auction where the bidding initial price decreases automatically with time following a parameterized curve until a bid is placed, at this point the bidder wins and the auction is closed.
+
+#### Goals
+Establish a mechanism to determine the price to become a validator.
+
+#### Non Goals
+The mechanism to become a validator.
+
+#### Assumptions
+Values defined at deployment are defined as a guide and can be modified to adapt the price curve to required results.
+
+## Specification
+
+#### Overview
+The solution consists of a Smart Contract that provides the following functionality:
+##### Start an auction
+This operation allows to start an auction by calculating the initial and final prices and registering the current block number as auction's start block, upon execution, an event with all auction details will be emitted.
+##### Get current bidding price
+This operation calculates the current bidding price using a specific price curve and the number of blocks mined between current block and the registered start block.
+##### Bid for current auction price
+This operation emits an event with address of the winner (sender) and auction details
+
+#### Data
+These variables will govern the price curve, the values are set at deployment
+| Descriptor | Constructor Value |
+| ----------- | ----------- |
+| Start Price  | 1000000 * 10 ** 18 |
+| Decay | 16 |
+| Scale Parameter | 100 |
+
+The cost to add a validator is calculated as follows:
+* ETHDKG Single Validator Cost (Two ETHDKG units operations 1.2M gas units) -> 1200000 * 2 * gasPriceInWeis
```

Comment:
Is there ever a conceivable situation that final cost is > starting cost given improbably, yet still possibly, high gas prices?

Reply:
I think that what @diligentis is pointing out is that as the start price is specified fixed at deployment but the final price is calculated based (among other things) on gas prices at auction start, it could happen that (because of extremely high gas prices) final price ends up being higher than start price. To fix this I'll be adding a condition to revert the auction start if it is not met.

Reply:
Added